src/sql/src/plan/statement/ddl.rs

```diff
@@ -1067,25 +1066,20 @@ fn get_encoding_inner<T: sql_parser::ast::AstInfo>(
             let regex = Regex::new(&regex)?;
             DataEncoding::Regex(RegexEncoding { regex })
         }
-        Format::Csv {
-            header_row,
-            n_cols,
-            delimiter,
-        } => {
-            let n_cols = if col_names.is_empty() {
-                match n_cols {
-                    Some(n) => *n,
-                    None => bail!(
-                        "Cannot determine number of columns in CSV source; specify using \
-                             CREATE SOURCE...FORMAT CSV WITH X COLUMNS"
-                    ),
+        Format::Csv { columns, delimiter } => {
+            let columns = match columns {
+                CsvColumns::Header { names } => {
+                    if names.is_empty() {
+                        bail!("[internal error] column spec should get names in purify")
```

Comment:
Should we panic here instead of bailing?

Reply:
We could, but IMO if we can avoid crashing customer's instances due to our bugs in instances where we know it's impossible to generate incorrect data we should do so. In this case the bail means we don't construct the dataflow, so we can't produce any incorrect values, so this is nicer.

Reply:
OK, fair enough